tests/lit-tests/func_template_shortvec_10.ispc

```diff
@@ -0,0 +1,24 @@
+// The test checks that we can do casts to pointers to short vectors inside templates (#3073).
+// RUN: %{ispc} %s --ast-dump --target=host --nostdlib | FileCheck %s
+// CHECK: unmasked /*unbound*/ T<4>(uniform int8 * uniform a)] "foo1"
+// CHECK: TypeCastExpr {{.*}} [/*unbound*/ T<4> * /*unbound*/]
+// CHECK: (instantiation <varying int32>) {{.*}}  [ unmasked varying int32<4>(uniform int8 * uniform a)] "foo1"
+// CHECK: TypeCastExpr {{.*}} [varying int32<4> * varying]
+
+// CHECK: unmasked /*unbound*/ T<N>(uniform int8 * uniform a)] "foo2"
+// CHECK: TypeCastExpr {{.*}} [/*unbound*/ T<N> * /*unbound*/]
+// CHECK: (instantiation <varying int32, 4>) {{.*}}  [ unmasked varying int32<4>(uniform int8 * uniform a)] "foo2"
+// CHECK: TypeCastExpr {{.*}} [varying int32<4> * varying]
+template <typename T> noinline unmasked T<4> foo1(uniform int8 *uniform a) {
+    uniform T<4> *uniform ptr1 = (uniform int<4> *uniform)a;
+    T<4> *ptr2 = (T<4> *)a;
```

Comment:
The template function is declared to return `unmasked T<4>` value, but has not return statement... Is it intended?

Reply:
No, it wasn't intentional, I fixed it.